smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -584,14 +589,20 @@ private void parseMixins(ShapeId id) {
         }
 
         ws();
+        expect('[');
+        ws();
+
         do {
+            clearPendingDocs();
```

Comment:
Do you need to clear pending docs in each iteration? Couldn't you do that at the end, once?

Reply:
This was an artifact of when I needed to do more backtracking.